src/gcp.rs

```diff
@@ -199,24 +199,28 @@ impl GcpAuthenticator {
         match self {
             Self::Credentials(inner) => {
                 let expires_in = inner.token.read().unwrap().expires_in() as u64;
-                let mut deadline =
-                    Duration::from_secs(expires_in - METADATA_TOKEN_EXPIRY_MARGIN_SECS);
+                let mut deadline = Duration::from_secs(
+                    expires_in
+                        .saturating_sub(METADATA_TOKEN_EXPIRY_MARGIN_SECS)
+                        .max(METADATA_TOKEN_ERROR_RETRY_SECS),
+                );
                 loop {
+                    debug!(
+                        %deadline, "Sleeping before refreshing GCP authentication token",
+                    );
```

Comment:
Since this calculation now happens both before and at the end of the loop, maybe make `expires_in` mutable and move the `deadline` calculation inside the loop.